### ISSUE TITLE
Sync branch while leaving uncommited files

### ISSUE THREAD

**Issue by ADogan** [baldwindavid/github-for-mac-issues#16]
You can only sync branches if you have no uncommitted files(that are not ignored and have been changed).

Request for change: syncing without committing all changed files.

**Comment by dv** [comment]
Please change this. I bumped into this seemingly arbitrary limitation on my first commit-sync. I had to go back to CLI and do a manual "git push".

**Comment by graygilmore** [comment]
This would be great, a super important feature.

**Comment by rfelix2121** [comment]
Yes, this is a pretty ridiculous limitation

**Comment by joshferrara** [comment]
Agreed. Would be very helpful to have this taken care of.

**Comment by InfernoZeus** [comment]
Likewise, can't believe this bug has been around for 9 months.

**Comment by ghostsquad** [comment]
Agreed.. this is completely ridiculous limitation. Not only that but many other source control applications allow you to push/pull regardless of what's checked out..

**Comment by FrancoisBlavoet** [comment]
Please fix this !

**Comment by Nixarn** [comment]
This. I'd really like it as well. Now I have to copy the content of a file, discard changed, sync, paste file content back

**Comment by SableRaf** [comment]
Actually, you can also push or pull specific changes using Command+P and Command+Shift+P

**Comment by craighooghiem** [comment]
Would be really nice to know that Command Shift P does this.. there's no UI for it at all, it just happens when you hammer the keys.

**Comment by karlnotturno** [comment]
Definitely this. I've been dealing with this for a while.
